src/Controller/LoginController.php

```diff
@@ -0,0 +1,438 @@
+        if (!\is_array($sessionTicket) || $forceAuthn) {
+            $sessionTicket = $this->ticketFactory->createSessionTicket($session->getSessionId(), $sessionExpiry);
+            $this->ticketStore->addTicket($sessionTicket);
+        }
+
+        /*
+         *  We are done. REDIRECT TO LOGGEDIN
+         * */
+        if (!isset($serviceUrl) && $this->authProcId === null) {
+            $loggedInUrl = Module::getModuleURL('casserver/loggedIn');
+            return new RunnableResponse(
+                [$this->httpUtils, 'redirectTrustedURL'],
+                [$loggedInUrl, $this->postAuthUrlParameters],
+            );
+        }
+
+        // Get the state.
+        $state = $this->getState();
+        $state['ReturnTo'] = $returnToUrl;
+        if ($this->authProcId !== null) {
+            $state[ProcessingChain::AUTHPARAM] = $this->authProcId;
+        }
+        // Attribute Handler
+        $mappedAttributes = $this->attributeExtractor->extractUserAndAttributes($state);
+        $serviceTicket = $this->ticketFactory->createServiceTicket([
+                                                                 'service' => $serviceUrl,
+                                                                 'forceAuthn' => $forceAuthn,
+                                                                 'userName' => $mappedAttributes['user'],
+                                                                 'attributes' => $mappedAttributes['attributes'],
+                                                                 'proxies' => [],
+                                                                 'sessionId' => $sessionTicket['id'],
+                                                             ]);
+        $this->ticketStore->addTicket($serviceTicket);
+
+        // Check if we are in debug mode.
+        if ($debugMode !== null && $this->casConfig->getOptionalBoolean('debugMode', false)) {
+            return $this->handleDebugMode($request, $debugMode, $serviceTicket);
+        }
+
+        $ticketName = $this->calculateTicketName($service);
+        $this->postAuthUrlParameters[$ticketName] = $serviceTicket['id'];
+
+        // GET
+        if ($redirect) {
+            return new RunnableResponse(
+                [$this->httpUtils, 'redirectTrustedURL'],
+                [$serviceUrl, $this->postAuthUrlParameters],
+            );
+        }
+        // POST
+        return new RunnableResponse(
+            [$this->httpUtils, 'submitPOSTData'],
+            [$serviceUrl, $this->postAuthUrlParameters],
+        );
+    }
+
+    /**
+     * @param   Request      $request
+     * @param   string|null  $debugMode
+     * @param   array        $serviceTicket
+     *
+     * @return XmlResponse
+     */
+    public function handleDebugMode(
+        Request $request,
+        ?string $debugMode,
+        array $serviceTicket,
+    ): XmlResponse {
+        // Check if the debugMode is supported
```

Comment:
I think this should return html with a `<pre> </pre>` block with the xml inside. debug mode is to show a human what the xml would look like, and returning actual xml makes the browser try to interpret it. I attached two screen shots: what it looks like with this PR and what the old system looked like (the users were different, so the xml content differs).
<img width="759" alt="Screenshot 2025-01-22 at 8 00 22 AM" src="https://github.com/user-attachments/assets/e9deb70b-0e59-41b1-b0bf-2197a1d364a9" />
<img width="892" alt="Screenshot 2025-01-22 at 8 00 59 AM" src="https://github.com/user-attachments/assets/bb65919f-2e96-47a0-98b9-fa3c402e27ec" />

Reply:
done